File: tests/fixtures/semgrep-offline/negative/Safe.java
import java.security.MessageDigest;
import java.security.SecureRandom;
import javax.crypto.Cipher;

public class Safe {
    void commandWithArgv(String arg) throws Exception {
        new ProcessBuilder(new String[] {"/usr/bin/git", "status", arg}).start();
    }
    MessageDigest strongHash() throws Exception { return MessageDigest.getInstance("SHA-256"); }
    Cipher strongCipher() throws Exception { return Cipher.getInstance("AES/GCM/NoPadding"); }
    SecureRandom unpredictable() { return new SecureRandom(); }
}

File: tests/fixtures/semgrep-offline/negative/safe.go
package main

import (
	"crypto/aes"
	"crypto/rand"
	"crypto/sha256"
	"crypto/tls"
	"os/exec"
)

func commandWithArgv(arg string) { exec.Command("/usr/bin/git", "status", arg) }
func strongHash(data []byte) []byte {
	h := sha256.New()
	return h.Sum(data)
}
func strongCipher(key []byte)          { aes.NewCipher(key) }
func tlsVerified() *tls.Config         { return &tls.Config{MinVersion: tls.VersionTLS13} }
func unpredictable(b []byte) (int, error) { return rand.Read(b) }
